web/impact/impact/permissions/v1_api_permissions.py

```diff
@@ -64,15 +76,24 @@ def has_permission(self, request, view):
 
 class OfficeHourPermission(IsAuthenticated):
     def has_permission(self, request, view):
+        self.message = self.get_message(request)
         roles = [UserRole.MENTOR, UserRole.AIR]
         return super().has_permission(request, view) and (
-                is_employee(request.user) or
-                request.user.programrolegrant_set.filter(
-                    program_role__user_role__name__in=roles,
-                    program_role__program__program_status='active',
-                ).exists())
+            is_employee(request.user) or
+            request.user.programrolegrant_set.filter(
+                program_role__user_role__name__in=roles,
+                program_role__program__program_status='active',
+            ).exists())
 
     def has_object_permission(self, request, view, office_hour):
         is_reserved = office_hour.finalist
         return (is_employee(request.user) or
                 office_hour.mentor == request.user and not is_reserved)
+
+    def get_message(self, request):
+        if request.method == 'POST':
```

Comment:
In general, data lookup should be done with a dict, eg 
```
{'POST': CREATE_PERMISSION_DENIED_DETAIL,
'PATCH': EDIT_PERMISSION_DENIED_DETAIL}
```

(use .get, with DEFAULT_PERMISSION_DENIED_DETAIL as default)